File: 09-DesignPatterns-Java/Solution/Adapter/src/Fly/TurkeyFlyStrategy.java
package Fly;

import Turkeys.Turkey;

public final class TurkeyFlyStrategy implements IFlyStrategy {
    private final Turkeys.Turkey _turkey;

    public TurkeyFlyStrategy(Turkey turkey) {
        _turkey = turkey;
    }

    @Override
    public void fly() {
            for (var i = 1; i <= 5; i++)
                _turkey.flutter();
    }
}

File: 10-DesignPatterns-CSharp/Solution/Adapter/Fly/TurkeyFlyStrategy.cs
namespace Adapter.Fly
{
    using Adapter.Turkeys;

    internal sealed class TurkeyFlyStrategy : IFlyStrategy
    {
        private readonly Turkey _turkey;

        internal TurkeyFlyStrategy(Turkey turkey)
        {
            _turkey = turkey;
        }

        public void Fly()
        {
            for (var i = 1; i <= 5; i++)
                _turkey.Flutter();
        }
    }
}
